tests/test_inner.py

```diff
@@ -755,16 +867,28 @@ def test_workflow_docker_build_error():
     """
     This is a test for what happens when the docker build fails.
     """
-
+    flexmock(DockerfileParser, content='df_content')
     this_file = inspect.getfile(PreRaises)
     mock_docker()
     fake_builder = MockInsideBuilder(failed=True)
     flexmock(InsideBuilder).new_instances(fake_builder)
+    watch_pre = Watcher()
+    watch_buildstep = Watcher(raise_exc=True)
     watch_prepub = Watcher()
     watch_post = Watcher()
     watch_exit = Watcher()
 
     workflow = DockerBuildWorkflow(MOCK_SOURCE, 'test-image',
+                                   prebuild_plugins=[{'name': 'pre_watched',
+                                                      'args': {
+                                                          'watcher': watch_pre
+                                                      }}],
+                                   buildstep_plugins=[{'name': 'buildstep_watched',
+                                                       'args': {
+                                                           'watcher': watch_buildstep,
+                                                       },
+                                                       'is_allowed_to_fail': False
```

Comment:
See comment elsewhere -- I think this should be the default for BuildStep plugins.